src/main/java/com/mraof/minestuck/computer/editmode/ServerEditHandler.java

```diff
@@ -400,13 +416,29 @@ else if(!isBlockItem(stack.getItem()) || !GristHelper.canAfford(event.getLevel()
 	@SubscribeEvent(priority=EventPriority.NORMAL)
 	public static void onLeftClickBlockControl(PlayerInteractEvent.LeftClickBlock event)
 	{
-		if(!event.getLevel().isClientSide && getData(event.getEntity()) != null)
+		if(!event.getLevel().isClientSide && event.getEntity() instanceof ServerPlayer player && getData(event.getEntity()) != null)
 		{
+			IEditTools cap = player.getCapability(MSCapabilities.EDIT_TOOLS_CAPABILITY).orElseThrow(() -> new IllegalStateException("EditTools Capability is empty in LeftClickBlock event on the server!"));
+			if(!event.getEntity().canInteractWith(event.getPos(), 0.0) || cap.isEditDragging())
+			{
+				event.setCanceled(true);
+				return;
+			}
+			
 			EditData data = getData(event.getEntity());
 			BlockState block = event.getLevel().getBlockState(event.getPos());
+			ItemStack stack = block.getCloneItemStack(null, event.getLevel(), event.getPos(), event.getEntity());
+			DeployEntry entry = DeployList.getEntryForItem(stack, data.connection, event.getLevel());
 			if(block.getDestroySpeed(event.getLevel(), event.getPos()) < 0 || block.getMaterial() == Material.PORTAL
-					|| (GristHelper.getGrist(event.getEntity().level, data.connection.getClientIdentifier(), GristTypes.BUILD) <= 0 && !MinestuckConfig.SERVER.gristRefund.get()))
+				|| (GristHelper.getGrist(event.getEntity().level, data.connection.getClientIdentifier(), GristTypes.BUILD) <= 0 && (!MinestuckConfig.SERVER.gristRefund.get() && (entry != null && entry.getCategory() != DeployList.EntryLists.ATHENEUM))))
```

Comment:
```suggestion
				|| (GristHelper.getGrist(event.getEntity().level, data.connection.getClientIdentifier(), GristTypes.BUILD) <= 0 && !MinestuckConfig.SERVER.gristRefund.get()
				|| entry == null || entry.getCategory() == DeployList.EntryLists.ATHENEUM)))
```
This is a little closer to its client-side counterpart. While there may still be some discrepancies between client-side and server-side editmode handlers, that can be a problem for after this PR.